### ISSUE TITLE
Snapcast Shairport configuration Allow MQTT (Or config from file)

### ISSUE THREAD

**Issue by dejanzelic** [badaix/snapcast#619]
Currently, the Shairport configuration (as far as I can tell) only allows the user to configure the `devicename` and `port`. I'm attempting to automate switching of the source on Snapcast depending on what's playing (Spotify, Mopidy, or Shairport). I can retrieve the status from Spotify using the API and Mopidy using MPD. However, I can not retrieve the status of Shairport. Shairport does support MQTT (https://github.com/idcrook/shairport-sync-mqtt-display/wiki/Build-shairport-sync-with-MQTT-support) However, there is no way to configure it using the Snapcast configuration:

`snapserver -s "airplay:///shairport-sync?name=Airplay[&devicename=Snapcast][&port=5000]"`

It would be nice to be able to optionally configure a MQTT server or load the configuration via a file (where I can set the MQTT server.

**Comment by badaix** [comment]
looks like a matter of the configuration in `/etc/shairport-sync.conf` and not like a command line setting.

**Comment by dejanzelic** [comment]
Thanks for the quick reply @badaix! Would snapcast still have shairport read from `/etc/shairport-sync.conf` even though it loads shairport via command line? I've tried to load the mqtt settings within `/etc/shairport-sync.conf` with no luck. Having snapcast read from the config file would be ideal.

**Comment by badaix** [comment]
why should snapcast read from the shairport-sync.conf file? Snapcast just launches shairport-sync. Is there any command line argument that needs to be passed to it?

**Comment by dejanzelic** [comment]
I worded that poorly but I think I've found the answer to my question. What I meant to ask was: will shairport-sync still read from the `shairport-sync.conf` file when started with Snapcast? Snapcast does not pass in the `-c` flag to Shairport, so I assumed that no configuration file was being read. However, after reading the man pages for Shairport, it states:

```
-c filename | --configfile=filename
Read configuration settings from filename. The default is to read them from the shairport-sync.conf in the System Configuration Directory -- /etc in Linux, /usr/local/etc in BSD unixes. For information about configuration settings, see the "Configuration File Settings" section above.
```

So even without the `-c` flag, Shairport will load the `shairport-sync.conf` file. I haven't set up what I'm looking for yet, but I've realized that this isn't a Snapcast issue. For this reason, I'm closing this issue. Thank you for the help @badaix!
